plutus-use-cases/src/Language/Plutus/Tutorial.lhs

```diff
@@ -0,0 +1,159 @@
+
+We can encode the two possible actions in a data type:
+
+> data CampaignAction = Collect | Refund
+>   deriving (Generic)
+> PlutusTx.makeLift ''CampaignAction
+
+The `CampaignAction` will be submitted as the redeemer script. Now we need one final bit of information, namely the identity (public key) of each contributor, so that we know the recipient of the refund. This public key is going to be stored in the data script of the contribution. 
+
+> data Contributor = Contributor Runtime.PubKey
+>   deriving (Generic)
+> PlutusTx.makeLift ''Contributor
+
+Now that we know the types of data and redeemer scripts, we automatically know the signature of the validator script:
+
+> type CampaignValidator = CampaignAction -> Contributor -> Runtime.PendingTx' -> ()
+
+`CampaignValidator` is a function that takes three parameters -- `CampaignAction`, `Contributor`, and `Runtime.PendingTx'` and produces a unit value `()` or fails with an error.
+
+If we want to implement `CampaignValidator` we need to know the parameters of the campaign, so that we can check if the selected `CampaignAction` is allowed. In Haskell we can do this by writing a function that takes a `Campaign` and produces a `CampaignValidator`:
+
+> mkValidator :: Campaign -> CampaignValidator
+
+However, we can't implement `mkValidator` right here, because we need to wrap it in Template Haskell quotes so that it can be compiled to Plutus Core. We therefore say
+
+> mkValidator = undefined
+
+and instead define a function that produces the validator script:
+
+> mkValidatorScript :: Campaign -> Validator
+> -- TODO: Rename Validator -> ValidatorScript ??
+> mkValidatorScript campaign = Validator val where
+>   val = UTXO.applyScript inner (UTXO.lifted campaign)
+>   -- ^ val is the obtained by applying `inner` to the lifted `campaign` value
+>   inner = UTXO.fromPlcCode $$(PlutusTx.plutus [|| 
+
+Anything between the `[||` and `||]` quotes is going to be _on-chain code_ and anything outside the quotes is _off-chain code_. We can now implement a lambda function that looks like `mkValidator`, starting with its parameters:
+
+>               \(c :: Campaign) (act :: CampaignAction) (con :: Contributor) (p :: Runtime.PendingTx') ->
+
+Before we check whether `act` is permitted, we define a number of intermediate values that will make the checking code much more readable. These definitions are placed inside a `let` block, which is closed by a corresponding `in` below.
+
+>               let
+>                   infixr 3 &&
+>                   (&&) :: Bool -> Bool -> Bool
+>                   (&&) = $$(Runtime.and)
+
+There is no standard library of functions that are automatically in scope for on-chain code, so we need to import the ones that we want to use from the `Runtime` module using the `\$\$()` splicing operator.
```

Comment:
I wonder if it would be clearer to just use `Runtime.and` directly, rather than having this operator definition with fixities etc.

Also, I think some of the things in `Runtime` maybe belong in `PlutusTx.Prelude` now?

Reply:
The problem is that `$$(Runtime.and)` doesn't work with the infix ticks. And I don't want to have to use it prefix with a lot of brackets. So instead of the operator definition we would have

```
and :: Bool -> Bool -> Bool
and = $$(Runtime.and)
```

and then use `and` with the infix ticks.